--- virt-manager-0.9.0/src/virtManager/console.py	2012-05-03 10:30:57.880351504 -0400
+++ foo/src/virtManager/console.py	2012-05-03 10:28:09.809796557 -0400
@@ -371,6 +371,7 @@ class VNCViewer(Viewer):
         ignore = user
 
         if not socketpath:
+            logging.debug("VNC connecting to %s:%s" % (host, port))
             self.display.open_host(host, port)
             return
 
@@ -497,7 +498,6 @@ class SpiceViewer(Viewer):
         ignore = socketpath
 
         uri = "spice://"
-        uri += (user and str(user) or "")
         uri += str(host) + "?port=" + str(port)
         logging.debug("spice uri: %s" % uri)
 
@@ -1087,7 +1087,7 @@ class vmmConsolePages(vmmGObjectUI):
 
             self.set_enable_accel()
 
-            if transport in ("ssh", "ext"):
+            if (transport in ("ssh", "ext")) and gaddr == "127.0.0.1":
                 if self.tunnels:
                     # Tunnel already open, no need to continue
                     return
@@ -1099,8 +1099,10 @@ class vmmConsolePages(vmmGObjectUI):
                     self.viewer.open_fd(fd)
 
             else:
-                self.viewer.open_host(connhost, connuser,
-                                      str(gport), gsocket)
+                host = gaddr
+                if gaddr == "127.0.0.1" or gaddr == "0.0.0.0":
+                    host = connhost
+                self.viewer.open_host(host, connuser, str(gport), gsocket)
 
         except Exception, e:
             logging.exception("Error connection to graphical console")
diff -rup virt-manager-0.9.0/src/virtManager/domain.py foo/src/virtManager/domain.py
--- virt-manager-0.9.0/src/virtManager/domain.py	2012-05-03 10:30:57.874352271 -0400
+++ foo/src/virtManager/domain.py	2012-05-03 10:25:18.423664662 -0400
@@ -903,7 +903,7 @@ class vmmDomain(vmmLibvirtObject):
             if gport != None:
                 gport = int(gport)
             gtype = gdev.type
-            gaddr = "127.0.0.1"
+            gaddr = gdev.listen or "127.0.0.1"
             gsocket = gdev.socket
 
         if connhost == None:
